io/src/Register.hpp

```diff
@@ -28,7 +28,7 @@
 #include <boost/serialization/serialization.hpp>
 #include <boost/serialization/nvp.hpp>
 
-#if __cplusplus >= 201103L
+#if SICONOS_CXXVERSION >= 201103L
```

Comment:
I did this 3.5 years ago (thanks git blame for reminding me this), so I had to take a stab at it and try to remember ... So looking at the code, ser_shared_ptr.hpp does the serialization of the std shared_ptr. Hence, I would add 
#if defined(SICONOS_STD_SHARED_PTR) && !defined(SICONOS_USE_BOOST_FOR_CXX11)
here.